src/v/cluster/cluster_discovery.cc

```diff
@@ -142,8 +148,41 @@ ss::future<bool> cluster_discovery::dispatch_node_uuid_registration_to_seeds(
     co_return false;
 }
 
-bool cluster_discovery::is_cluster_founder() const {
-    return config::node().seed_servers().empty();
+/*static*/ std::optional<node_id>
+cluster_discovery::get_cluster_founder_node_id() {
+    if (config::node().empty_seed_starts_cluster()) {
+        if (config::node().seed_servers().empty()) {
+            return node_id{0};
+        }
+        return {};
+    } else {
+        if (auto idx = get_node_index_in_seed_servers(); idx) {
+            return node_id{*idx};
+        }
+        return {};
+    }
+}
+
+/*static*/ std::optional<int32_t>
+cluster_discovery::get_node_index_in_seed_servers() {
+    const std::vector<config::seed_server>& seed_servers
+      = config::node().seed_servers();
+    vassert(
```

Comment:
do we have a configuration validator that will error out gracefully before reaching this assertion ?